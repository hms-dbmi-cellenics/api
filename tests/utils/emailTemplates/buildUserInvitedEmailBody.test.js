const buildUserInvitedEmailBody = require('../../../src/utils/emailTemplates/buildUserInvitedEmailBody');

const testUserEmail = 'test@email.com';
const mockInviterUser = { email: 'inviter@example.com' };
const mockExperimentId = 'mock-experiment-id';

process.env.DOMAIN_NAME = 'localhost.test';

describe('buildUserInvitedEmailBody', () => {
  it('Should build the correct message', () => {
    const emailParams = buildUserInvitedEmailBody(testUserEmail, mockExperimentId, mockInviterUser);
    expect(emailParams).toMatchSnapshot();
  });
});
