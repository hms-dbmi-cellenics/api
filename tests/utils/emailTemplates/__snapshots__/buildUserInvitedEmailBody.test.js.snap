// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildUserInvitedEmailBody Should build the correct message 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "test@email.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <p>
              Hello, <br/>
              Your collaborator inviter@example.com has invited you to explore their single cell RNA-seq project in Cellenics. <br/><br/>
              Access it with the link below: <br/>
              <a href=\\"https://test.com/experiments/mock-experiment-id/data-exploration\\">https://test.com/experiments/mock-experiment-id/data-exploration</a> <br/> <br/>
              Cellenics is a user-friendly online tool for single cell RNA-seq data analysis. <br/>
              The platform is designed specifically for biologists, and it's completely free for academic researchers.<br/><br/><div>test hi info</div>
              Best Regards, <br/>
              Cellenics team
            <p/>
        </body>
      </html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <p>
              Hello, <br/>
              Your collaborator inviter@example.com has invited you to explore their single cell RNA-seq project in Cellenics. <br/><br/>
              Access it with the link below: <br/>
              <a href=\\"https://test.com/experiments/mock-experiment-id/data-exploration\\">https://test.com/experiments/mock-experiment-id/data-exploration</a> <br/> <br/>
              Cellenics is a user-friendly online tool for single cell RNA-seq data analysis. <br/>
              The platform is designed specifically for biologists, and it's completely free for academic researchers.<br/><br/><div>test hi info</div>
              Best Regards, <br/>
              Cellenics team
            <p/>
        </body>
      </html>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Invitation to join a project in Cellenics",
    },
  },
  "Source": "some@email.com",
}
`;
